### PR TITLE
fix(ThreadMember): remove audit log reason parameter

### PR DESCRIPTION
**Pull request by Renegade334**
**Please describe the changes this PR makes and why it should be merged:**
The [Add Thread Member](https://discord.com/developers/docs/resources/channel#add-thread-member) and [Remove Thread Member](https://discord.com/developers/docs/resources/channel#remove-thread-member) endpoints do not create audit log entries, and therefore do not accept the `X-Audit-Log-Reason` header. 

**Status and versioning classification:**

- Code changes have been tested against the Discord API, or there are no code changes
- I know how to update typings and have done so, or typings don't need updating
- This PR includes breaking changes (methods removed or renamed, parameters moved or removed)

### CODE REVIEW

**Review by jaw0r3k — APPROVED** [review]

**Review by kyranet — APPROVED** [review]

File: packages/discord.js/src/structures/ThreadMember.js [hunk 1]
```diff
@@ -101,11 +101,10 @@ class ThreadMember extends Base {
 
   /**
    * Removes this member from the thread.
-   * @param {string} [reason] Reason for removing the member
-   * @returns {ThreadMember}
+   * @returns {Promise<ThreadMember>}
```
Comment (Jiralite) [inline packages/discord.js/src/structures/ThreadMember.js:104]: This could be merged immediately in a separate pull request without breaking changes. Do you mind doing that?

Comment (Renegade334) [reply packages/discord.js/src/structures/ThreadMember.js:104]: Bien sûr.

**Comment by Jiralite** [comment]
Please resolve conflicts!

**Comment by Renegade334** [comment]
Will rebase when I get the chance. 👍

**Review by vladfrangu — APPROVED** [review]

**Review by iCrawl — APPROVED** [review]

**Review by Jiralite — APPROVED** [review]
